src/builtins/builtin.ts

```diff
@@ -0,0 +1,48 @@
+import { Segment } from 'memory/memory';
+import { Relocatable } from 'primitives/relocatable';
+import { SegmentValue } from 'primitives/segmentValue';
+
+/** Abstract class representing builtin runners */
+export abstract class BuiltinRunnerBase {
```

Comment:
I don't really understand why we need a dedicated builtin runner. It seems to me that we can just add rules to segments getter, default segments being non-builtins segments just return the value of the underlying array, and builtins segments apply the rule their. Consequently, from the VM run point of view, builtins are transparent